static/app/utils/metrics/useMetricsMeta.tsx

```diff
@@ -58,6 +61,65 @@ export function useMetricsMeta(
   };
 }
 
+/**
+ * Like useMetricsMeta, but it maps extracted custom metrics into a separate namespace
+ */
+export const useVirtualizedMetricsMeta = (
+  pageFilters: Partial<PageFilters>,
+  useCases: UseCase[] = DEFAULT_USE_CASES,
+  filterBlockedMetrics = true,
+  enabled: boolean = true
+): {
+  data: MetricMeta[];
+  isLoading: boolean;
+  isRefetching: boolean;
+  refetch: () => void;
+} => {
+  const {
+    getVirtualMRI,
+    getVirtualMeta,
+    isLoading: isVirtualMetricsContextLoading,
+  } = useVirtualMetricsContext();
+
+  const {data, isLoading, isRefetching, refetch} = useMetricsMeta(
+    pageFilters,
+    useCases,
+    filterBlockedMetrics,
+    enabled
+  );
+
+  const newMeta = useMemo(() => {
+    const virtualMetrics = new Set<MRI>();
+    // Filter all extracted custom metrics and map them to virtual metrics
+    const otherMetrics = data.filter(meta => {
+      if (!isExtractedCustomMetric(meta)) {
+        return true;
+      }
+
+      const virtualMRI = getVirtualMRI(meta.mri);
+      // If there is no virtual MRI, we don't want to show this metric
+      if (!virtualMRI) {
+        return false;
+      }
```

Comment:
As the stub implementation of `getVirtualMRI` always returns `null`, this will have the effect of hiding internal extracted metrics from the UI wherever it is used.